### PR TITLE
Post about GLPI using atoum

### PR DESCRIPTION
**Pull request by Grummfy**
I finally got time to do it!

### CODE REVIEW

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "GLPI migrate to atoum"
+date: 2017-08-10 10:00:00
+author: "@Grummfy"
+categories: news
+---
+
+The [GLPI project][GLPI] [recently announced][announced-en][^1] that they have migrated to atoum for their unit test suite. So let's get back on
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: they have migrated their unit test suite to atoum.

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+this news. We ask several questions to the official team of the project to have some feedback, [Johan Cwiklinski](https://github.com/trasher)
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: We interviewed Johan Cwiklinski and some other members of the project official team about this migration.

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+The [GLPI project][GLPI] [recently announced][announced-en][^1] that they have migrated to atoum for their unit test suite. So let's get back on
+this news. We ask several questions to the official team of the project to have some feedback, [Johan Cwiklinski](https://github.com/trasher) 
+with some member of his team answered them.
+
+## 1. Can you summary the GLPI project and the usage of PHP inside it?
+
+GLPI is an open source IT asset management and Service Desk software. The project has started 15 years ago and is written in PHP.
+The project was led by an association until 2015, and since then the development is headed by [Teclib](www.teclib.com).
+
+Since 2015; we are trying to rewrite and modernise the whole project, starting with the backend.
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: modernize

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## 2. When did you start using unit test for the project?
+
+At first, one regular contributor has tried to add unit tests into the project in 2010. But project managers did not follow him, and this has quickly been abandoned.
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: Firstly
this was quickly abandoned.

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+Then, in 2015, the same contributor has tried again to add tests on the project; with much more success :)
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: The same contributor tried again

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+## 3. What were the issues you were facing with PHPUnit? Why did you choose atoum instead of other tools to resolve it?
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: What kind of difficulties do you encounter

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+* concurrent run of tests (even if it has been disabled for GLPI),
+* fully isolated tests,
+* chained calls,
+* more natural way to write tests (this is maybe just my point of view - but this is really simpler to me).
+
+Last but not least; atoum is a French project at the origin; just like GLPI!
+
+## 4. Did you meet some issue during the migration from PHPUnit to atoum?
+
+Well... We had to entirely rewrite our tests. It’s more a painful than a difficult work at first; but finally, we’ve not spent that much time on the rewrite process.
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: we havent spent too much time on this rewrite work.

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+Of course, we didn’t implement all atoum features; but we plan to update our tests in the future.
+
+We did have a really basic usage of PHPUnit; the only thing that required much more work was the `@depends`[^2] capability of PHPUnit.
+That was quite easy to change; and has benefited: we now run all our tests from a fresh, out of the box GLPI environment.
+
+Another issue was that atoum is much less permissive than PHPUnit per default; but this was not an issue because we got many little things fixed (such a PHP notices...).
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: this was not a real issue

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+## 5. Did you have any advice for other people that would like to migrate to atoum from PHPUnit?
+
+First of all, I’d advise people to write unit tests.
+
+Most people I know actually use PHPUnit because they’ve always used it, so let’s give atoum a try. ;) Well, at least try it; on a new project for example,
+or just on a small part of your code.
+
+## 6. You recently receive a price related to security, can you tel us more about it?
+
+Teclib has been involved in a project to adapt it to match SoC (Security operation centre) requirements. The project had also been an opportunity to develop a REST API.
```
Comment (mikaelrandy) [inline _posts/2017-08-10-glpi-atoum.markdown]: A link on Teclib word ?

**Review by mikaelrandy — COMMENTED** [review]
Seems a little bit too close to Google translation ;)

**Comment by Grummfy** [comment]
thanks for your review  @mikaelrandy

**Review by mikaelrandy — APPROVED** [review]

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 10]
```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title: "GLPI migrate to atoum"
+date: 2017-08-10 10:00:00
+author: "@Grummfy"
+categories: news
+---
+
+The [GLPI project][GLPI] [recently announced][announced-en][^1] that they have migrated their unit test suite to atoum. So let's get back on
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- suite
+ suites
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+this news. We interviewed [Johan Cwiklinski](https://github.com/trasher) and some other members of the project's official team about this migration.
+
+## 1. Can you summary the GLPI project and the usage of PHP inside it?
+
+[GLPI][GLPI] is an open source IT asset management and Service Desk software. The project has started 15 years ago and is written in PHP.
+The project was led by an association until 2015, and since then the development is headed by [Teclib](www.teclib.com).
+
+Since 2015; we are trying to rewrite and modernize the whole project, starting with the backend.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- 2015;
+ 2015,
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+## 2. When did you start using unit test for the project?
+
+Firstly, one regular contributor has tried to add unit tests into the project in 2010. But project managers did not follow him, and this was quickly abandoned.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- this
+ it
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+Then, in 2015, The same contributor tried again to add tests on the project; with much more success :)
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- Then, in 2015, The same
+ Forward to 2015, the same
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+## 3. What kind of difficulties do you encounter with PHPUnit? Why did you choose atoum instead of other tools to resolve it?
+
+We didn’t really face issues with PHPUnit. This is more a choice that was dictated by atoum capabilities; like:
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- choice
+ technical choice
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- dictated
+ led
```

(still not convinced by “led”, but much better than “dictated”).

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+* testing variable types (if you test an integer and get a string, this is not correct),
+* wonderful mocking system (that allows mocking PHP native functions, constants, ...),
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- , ...
+ …
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+* use closure to test outputs, exceptions, ... (way more interesting than phpunit’s annotations),
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- , ...
+ …
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- phpunit's
+ PHPUnit's
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+* concurrent run of tests (even if it has been disabled for GLPI),
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- tests
+ test cases
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+* fully isolated tests,
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- tests
+ test cases
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+* chained calls,
+* more natural way to write tests (this is maybe just my point of view - but this is really simpler to me).
+
+Last but not least; atoum is a French project at the origin; just like GLPI!
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- least;
+ least:
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+
+## 4. Did you meet some issue during the migration from PHPUnit to atoum?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- meet
+ face
```

```diff
- some issue
+ some issues
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+
+Well... We had to entirely rewrite our tests. It’s more a painful than a difficult work at first; but finally, we haven't spent too much time on this rewrite work.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- Well...
+ Well…
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- our tests
+ our test suites
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- a painful
+ painful
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]: Do we have some numbers here? Time *t* spent to rewrite *x* test suites of *y* test cases? Did they find new bugs? What about the execution time? Is it easier now to write new tests? Is it easier for newcomers to write or to maintain new test suites?

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+Of course, we didn’t implement all atoum features; but we plan to update our tests in the future.
+
+We did have a really basic usage of PHPUnit; the only thing that required much more work was the `@depends`[^2] capability of PHPUnit.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ; the only
+ . The only
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+That was quite easy to change; and has benefited: we now run all our tests from a fresh, out of the box GLPI environment.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ; and has benefited: we
+ and brought some nice benefits: We
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+
+Another issue was that atoum is much less permissive than PHPUnit per default; but this was not a real issue because we got many little things fixed (such a PHP notices...).
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ; but this
+ , but this
```

```diff
- such a PHP notices...
+ such as PHP notices…
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+
+## 5. Did you have any advice for other people that would like to migrate to atoum from PHPUnit?
+
+First of all, I’d advise people to write unit tests.
+
+Most people I know actually use PHPUnit because they’ve always used it, so let’s give atoum a try. ;) Well, at least try it; on a new project for example,
+or just on a small part of your code.
+
+## 6. You recently receive a price related to security, can you tel us more about it?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- tel
+ tell
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 26]
```diff
@@ -0,0 +1,82 @@
...
+
+[Teclib](www.teclib.com) has been involved in a project to adapt it to match SoC (Security operation centre) requirements. The project had also been an opportunity to develop a REST API.
+
+Being part of the OW2 Consortium community, GLPI received the 2017 Technology Council special prize for its work on addressing security requirements but also
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]: Add link to OW2 Consortium please.

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]: Add link to the news or the tweet, with at least the photo!

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+
+Most people I know actually use PHPUnit because they’ve always used it, so let’s give atoum a try. ;) Well, at least try it; on a new project for example,
+or just on a small part of your code.
+
+## 6. You recently receive a price related to security, can you tel us more about it?
+
+[Teclib](www.teclib.com) has been involved in a project to adapt it to match SoC (Security operation centre) requirements. The project had also been an opportunity to develop a REST API.
+
+Being part of the OW2 Consortium community, GLPI received the 2017 Technology Council special prize for its work on addressing security requirements but also 
+the big working currently being made to modernise the framework. The jury has mentioned the implementation of unit tests and the switch from PHPUnit to atoum !
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- atoum !
+ atoum!
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 28]
```diff
@@ -0,0 +1,82 @@
...
+
+## 7. Anything to add?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- Anything to add?
+ If you have anything else to say, please feel free!
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 29]
```diff
@@ -0,0 +1,82 @@
...
+
+I’ve been involved in various projects (open source or not) for years. As many developers, I was not really aware of unit testing.
+When I’ve started to write unit tests on my PHP applications, I chose PHPUnit (the only one I’d heard about at the time) but never really liked it.
+
+When I’ve discovered atoum (I don’t remember exactly when, but that was far before the first semver release); I’ve immediately loved it. That was much
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ;
+ ,
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 30]
```diff
@@ -0,0 +1,82 @@
...
+clearer to me to understand and write tests; I began to like that! ;) I’ve used atoum in several closed source projects (even when the "frameworks" say
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ;
+ .
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 31]
```diff
@@ -0,0 +1,82 @@
...
+to use something else); but also in [Galette](http://galette.eu)[^3]; which has unit tests powered by atoum since 2013.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- ;
+ .
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 32]
```diff
@@ -0,0 +1,82 @@
...
+
+
+
+## Final word
+I would like to thank Johan and his team for this feedback, it was really interesting and I hope you also like it. If you use atoum on your project,
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- this feedback
+ this valuable feedback
```

“it was really interesting” Why?

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 33]
```diff
@@ -0,0 +1,82 @@
...
+we would be happy to hear some feedback about it, it's always interesting.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- feedback
+ feedbacks
```

```diff
- interesting
+ important
```

**Review by Hywan — CHANGES_REQUESTED** [review]
Thanks for taking time for that!

**Comment by Grummfy** [comment]
thanks @Hywan for the review. 
For information I don't know which is good but antidote is saying that modernize is American and modernise is British... For me it's the same, but I let modernise, if you think it's important I will change it. Same for feedback it say that it's not countable ... 
So except for theses two points I have made the change reviewed.

**Comment by Hywan** [comment]
Modernise is English 🇬🇧, modernize is simplified English 🇺🇸.
Use English.

Yes, “feedbacks” is incorrect, I forgot, sorry.

Thanks for the modifications! Why so many “;” before?

**Comment by Grummfy** [comment]
Because the answer to the question was, what it was given to me by email, and there is a lot of ';'
So I just copy-paste the answer to my questions ;)

File: _posts/2017-08-10-glpi-atoum.markdown (hunk 14 above)
Comment (trasher) [reply _posts/2017-08-10-glpi-atoum.markdown]: What about "This is more a technical choice for atoum capabilities"?

**Comment by Grummfy** [comment]
@trasher an y extra comments?

**Comment by trasher** [comment]
I've just send you by mail extra info you requested; it's OK for me :)

**Comment by Grummfy** [comment]
SO I think it's ready : @Hywan &  @mikaelrandy I would like to have a last feedback from you

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 34]
```diff
@@ -0,0 +1,112 @@
...
+title: "GLPI migrate to atoum"
+date: 2017-08-10 10:00:00
+author: "@Grummfy"
+categories: news
+---
+
+The [GLPI project][GLPI] [recently announced][announced-en][^1] that they have migrated their unit test suites to atoum. So let's get back on
+this news. We interviewed [Johan Cwiklinski](https://github.com/trasher) and some other members of the project's official team about this migration.
+
+## 1. Can you summary the GLPI project and the usage of PHP inside it?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- summary
+ summarize
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- the usage of PHP inside it?
+ its usage of PHP?
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 35]
```diff
@@ -0,0 +1,112 @@
...
+
+[GLPI][GLPI] is an open source IT asset management and Service Desk software. The project has started 15 years ago and is written in PHP.
+The project was led by an association until 2015, and since then the development is headed by [Teclib](www.teclib.com).
+
+Since 2015, we are trying to rewrite and modernise the whole project, starting with the backend.
+
+## 2. When did you start using unit test for the project?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- unit test
+ unit tests
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 36]
```diff
@@ -0,0 +1,112 @@
...
+
+Firstly, one regular contributor has tried to add unit tests into the project in 2010. But project managers did not follow him, and it was quickly abandoned.
+Forward to 2015, the same contributor tried again to add tests on the project, with much more success :)
+
+## 3. What kinds of difficulties do you encounter with PHPUnit? Why did you choose atoum instead of other tools to resolve it?
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- kinds
+ kind
```

Comment (Hywan) [reply _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- do you
+ did you
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 37]
```diff
@@ -0,0 +1,112 @@
...
+* Concurrent run of test cases (even if it has been disabled for GLPI),
+* Fully isolated test cases,
+* Chained calls,
+* More natural way to write tests (this is maybe just my point of view - but this is really simpler to me).
+
+Last but not least: atoum is a French project at the origin, just like GLPI!
+
+## 4. Can you tell us why you have disabled the concurrent test for the GLPI project?
+
+In some case, due to the current GLPI code, some tests could not be run concurrently, this cause unpredictable results. Also, GLPI use a huge
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- Also, GLPI use
+ Also, GLPI uses
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 38]
```diff
@@ -0,0 +1,112 @@
...
+Of course, we didn’t implement all atoum features, but we plan to update our tests in the future.
+
+We did have a really basic usage of PHPUnit. The only thing that required much more work was the `@depends`[^2] capability of PHPUnit.
+That was quite easy to change and brought some nice benefits: We now run all our tests from a fresh, out of the box GLPI environment.
+
+Another issue was that atoum is much less permissive than PHPUnit per default, but this was not a real issue because we got many little things fixed (such as PHP notices…).
+
+Here is some telemetry information regarding these tests:
+* We had 25 tested classes with PHPUnit, and (of course!) still 25 with atoum,
+* There were about 2600 assertions with PHPUnit, but more than 5000 with atoum. This came for various reasons, one was that many returns were not tested and lend to un-understandable situations,
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- lend
+ led
```

?

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 39]
```diff
@@ -0,0 +1,112 @@
...
+* I’ve spent about *2 or 3 work days* to achieve the *complete migration*. I’ve also taken a few hours to migrate already opened pull requests that provided unit tests.
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- provided unit tests.
+ provided unit tests,
```

File: _posts/2017-08-10-glpi-atoum.markdown [hunk 40]
```diff
@@ -0,0 +1,112 @@
...
+* Since atoum migration, some tests have been added. We now test 66 classes with 9000 assertions.[^3] :)
```
Comment (Hywan) [inline _posts/2017-08-10-glpi-atoum.markdown]:
```diff
- assertions.[^3]
+ assertions,[^3]
```

**Review by Hywan — CHANGES_REQUESTED** [review]

**Comment by Grummfy** [comment]
@Hywan thanks, fixed ;)

**Pull request merged**